python/paddle/fluid/tests/unittests/test_requantize_mkldnn_op.py

```diff
@@ -0,0 +1,106 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import unittest
+import numpy as np
+from op_test import OpTest
+
+
+def format_reorder(out, size):
+    in_n = size[0]
+    out_h = size[2]
+    out_w = size[3]
+    out_c = size[1]
+    out_tmp = np.zeros((in_n, out_h, out_w, out_c))
+    for n in range(in_n):
+        for i in range(out_h):
+            for j in range(out_w):
+                for m in range(out_c):
+                    out_tmp[n, i, j, m] = out[n, m, i, j]
+    return out_tmp.reshape(in_n, out_c, out_h, out_w)
+
+
+class TestReQuantizeOp(OpTest):
+    def setUp(self):
+        self.op_type = 'requantize'
+        self.scale_in = 2.0
+        self.scale_out = 1.5
+        self.input_size = [1, 1, 5, 5]
+        self.data_type = 'int8'
+        self.set_scale()
+        self.set_data_type()
+
+        scale_shift = self.scale_out / self.scale_in
+
+        if self.data_type == 'int8':
+            input = (np.random.randint(0, 100, self.input_size) - 50
+                     ).astype(self.data_type)
+            output_tmp = np.round(input.astype('float32') *
+                                  scale_shift).astype('int8')
+            output = format_reorder(output_tmp, self.input_size)
+        else:
+            input = (np.random.randint(0, 100,
+                                       self.input_size)).astype(self.data_type)
+            output_tmp = np.round(input.astype('float32') *
+                                  scale_shift).astype('uint8')
+            output = format_reorder(output_tmp, self.input_size)
```

Comment:
line59 is the same with line53, move them out of `if-else` and remain one of it.

Reply:
Sure, I can do this.